components/feature/pwa/src/main/java/mozilla/components/feature/pwa/ext/Session.kt

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.pwa.ext
+
+import mozilla.components.browser.session.Session
+import mozilla.components.concept.engine.manifest.WebAppManifest.Icon.Purpose
+import kotlin.math.min
+
+private const val MIN_INSTALLABLE_ICON_SIZE = 192
+
+/**
+ * Checks if the current session represents an installable web app.
+ *
+ * Websites are installable if:
+ * - The site is served over HTTPS
+ * - The site has a valid manifest with a name or short_name
+ * - The icons array in the manifest contains an icon of at least 192x192
```

Comment:
I think having a [start_url](https://developers.google.com/web/fundamentals/web-app-manifest/#start-url) property should also be also a requirement to be installable. What do you think?

Reply:
`start_url` and `name` must be set if `webAppManifest` is not null. They aren't nullable so the manifest won't be parsed without them.

Reply:
Got you! Can they be empty String?  Or are we preventing that with an exception?

Reply:
We aren't currently checking that, but we should. The spec actually defines that name can be omitted, so I'm going to update this PR with some changes to the parser.

Reply:
👍